src/Avalonia.Controls/Application.cs

```diff
@@ -190,18 +194,25 @@ public IResourceDictionary Resources
         public WindowCollection Windows { get; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether this instance is existing.
+        /// Gets or sets a value indicating whether this instance is shutting down.
         /// </summary>
         /// <value>
-        ///   <c>true</c> if this instance is existing; otherwise, <c>false</c>.
+        ///   <c>true</c> if this instance is shutting down; otherwise, <c>false</c>.
         /// </value>
-        internal bool IsExiting { get; set; }
+        internal bool IsShuttingDown { get; private set; }
 
-        /// <summary>
-        /// Initializes the application by loading XAML etc.
-        /// </summary>
-        public virtual void Initialize()
+        public void Run()
```

Comment:
Right now this method is reentrant. One can stop an app loop and start it again.

I don't see why we are introducing an arbitrary limitation here.

Reply:
When run returns the app closes. How do you start it again then? We could introduce Pause() and Resume(). Maybe I don't understand this entirely.

Reply:
The limitation should be gone now